### DISCUSSION TITLE
p.268 Postman 401 unauthorised error & WebSecurityConfigurerAdapter deprecated

### DISCUSSION THREAD

**Discussion by Jay-yoon10** [fsoftwareengineer/todo-application#135] [질문과 답변]
안녕하세요,
현재 4장 인증 통합 마지막 부분을향해 가고있는데요.
WebSecurityConfigurerAdapter 가 더이상 지원이 안된다고 오류가 자꾸 떠서 
계속 찾아보고 새로운 방법으로 접근해보았습니다.

다만 postman 에서 test중에 URI 상관없이 (todo포함) 다 401 Unauthorized 라고 뜨네요..
Dependency 에서 spring security 부분을 제거하면 작동이 잘되는데 어떤게 문제일까요? 
혹시 spring security update 되고 나서 WebSecurityConfigurerAdapter 없이 새로운 방법으로 접근하신분 계시면 공유 부탁드리겠습니다 !

### **Basic-Auth**
<img width="944" alt="image" src="https://user-images.githubusercontent.com/93105607/198559028-357e6660-7c67-4cab-9879-1f2a66020011.png">

### **application.properties**
<img width="452" alt="image" src="https://user-images.githubusercontent.com/93105607/198559131-c4495ac4-701e-44c2-b3fb-eb3f158c9771.png">

### **http://localhost:8080/auth/signup**
<img width="660" alt="image" src="https://user-images.githubusercontent.com/93105607/198560224-11d183b6-443d-4643-8361-9710c1650b82.png">

### **WebSecurityConfig.java**
```java
import org.springframework.context.annotation.Bean;
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.web.filter.CorsFilter;
import toDoList.demo.security.Jwt.JwtAuthenticationFilter;

@Slf4j
//@Configuration
public class WebSecurityConfig  {

    @Autowired
    JwtAuthenticationFilter jwtAuthenticationFilter;
    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {

        http.cors() // already set cors configuration in WebMvcConfig
                .and()
                .csrf()// csrf is not being used
                .disable()
                .httpBasic()// token is used, so basic auth is disabled
                .disable()
                .sessionManagement()  // not session reliable
                .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                .and()
                .authorizeRequests() // path "/" and "/auth/**" no need to authorise
                .antMatchers("/","/auth/**").permitAll()
                .anyRequest() // any other paths besides path "/" and "/auth/**"  need to be authenticated.
                .authenticated();
//
        http.addFilterAfter(
                jwtAuthenticationFilter,
                CorsFilter.class
        );
        return http.build();
    }


}

**Comment by fsoftwareengineer** [comment]
build.gradle 파일을 첨부 해 주시기 바랍니다.

**Reply by Jay-yoon10** [reply]
안녕하세요.
책 잘보고 많이 배우고있습니다.
이번 프로젝트에서 Gradle대신에 maven 으로 진행 하였습니다.
혹시 가능하시다면 WebSecurityConfigurerAdapter 사용안하는 방법도 소개해주실 수 있으신가요?

### maven파일 



	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.7.5</version>
		<relativePath/> 
	</parent>
	<groupId>toDoList</groupId>
	<artifactId>demo</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>TodoList</name>
	<description>Demo project for Spring Boot</description>
	<properties>
		<java.version>17</java.version>
	</properties>
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-jpa</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-validation</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>

		<dependency>
			<groupId>com.mysql</groupId>
			<artifactId>mysql-connector-j</artifactId>
			<scope>runtime</scope>
		</dependency>
		<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
			<optional>true</optional>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.jsonwebtoken</groupId>
			<artifactId>jjwt</artifactId>
			<version>0.9.1</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-security</artifactId>
			<version>2.6.13</version>
		</dependency>
	</dependencies>

**Comment by kws9208** [comment]
혹시 해결하셨나요?

**Reply by kws9208** [reply]
아래 링크 참조하여 해결했습니다!
https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter
